### PR TITLE
Adding retrive member function and getMember function in LML.sol

### PR DESCRIPTION
**Pull request by nishantkshyp2004**

### CODE REVIEW

**Comment by inovizz** [comment]
We need to add test cases as well for both these functions.

File: contracts/LMS.sol [hunk 1]
```diff
@@ -82,6 +82,25 @@ contract LMS is Killable {
         memberIndex[account] = numMembers;
     }
 
+    function getMember(uint i) constant returns (string memberString){
```
Comment (nrchandan) [inline contracts/LMS.sol]: I don't understand the purpose of this function. Please write a short comment explaining the purpose (ala python docstrings).

File: contracts/LMS.sol [hunk 2]
```diff
@@ -82,6 +82,25 @@ contract LMS is Killable {
...
+
+    function retriveMembers() constant onlyMember returns (string memberString, uint8 count){
+
+        for (uint i=1; i<=numMembers; i++) {
+            memberString = memberString.toSlice().concat('|'.toSlice()).toSlice().concat(getMember(i).toSlice());
+            count++;
+        }
+
+    }
+
```
Comment (nrchandan) [inline contracts/LMS.sol]: Like Sanchit commented, write accompanying tests. I would recommend writing tests before writing the functions - the TDD/BDD approach.

File: contracts/LMS.sol [hunk 3]
```diff
@@ -82,6 +82,25 @@ contract LMS is Killable {
...
+    function retriveMembers() constant onlyMember returns (string memberString, uint8 count){
+
+        for (uint i=1; i<=numMembers; i++) {
+            memberString = memberString.toSlice().concat('|'.toSlice()).toSlice().concat(getMember(i).toSlice());
+            count++;
+        }
+
```
Comment (nrchandan) [inline contracts/LMS.sol]: Remove the empty line here and elsewhere.

File: contracts/LMS.sol [hunk 4]
```diff
@@ -82,6 +82,25 @@ contract LMS is Killable {
...
+        var parts = new strings.slice[](4);
+        parts[0] = members[i].name.toSlice();
+        parts[1] = StringLib.addressToString(members[i].account).toSlice();
+        parts[2] = StringLib.uintToString(uint(members[i].status)).toSlice();
+        parts[3] = StringLib.uintToString(uint(members[i].dateAdded)).toSlice();
+        memberString = ";".toSlice().join(parts);
+
+    }
+
+    function retriveMembers() constant onlyMember returns (string memberString, uint8 count){
```
Comment (nrchandan) [inline contracts/LMS.sol]: Spelling mistake - should be "retrieveMembers". Also, there should be a single space between closing parenthesis and opening brace.

**Comment by nrchandan** [comment]
You should also have reused/refactored the existing getMemberDetails function.

File: contracts/LMS.sol [hunk 5]
```diff
@@ -82,6 +82,25 @@ contract LMS is Killable {
         memberIndex[account] = numMembers;
     }
 
+    function getMember(uint i) constant returns (string memberString){
+        var parts = new strings.slice[](4);
+        parts[0] = members[i].name.toSlice();
+        parts[1] = StringLib.addressToString(members[i].account).toSlice();
+        parts[2] = StringLib.uintToString(uint(members[i].status)).toSlice();
+        parts[3] = StringLib.uintToString(uint(members[i].dateAdded)).toSlice();
+        memberString = ";".toSlice().join(parts);
```
Comment (nrchandan) [inline contracts/LMS.sol]: Refactor/reuse the getMemberDetails function.

**Pull request closed without merging**